app/api/export/index.js

```diff
@@ -0,0 +1,149 @@
+// @flow
+import moment from 'moment';
+import { sendFileToUser } from './utils';
+import type {
+  ExportTransactionsRequest,
+  ExportTransactionsResponse,
+} from '../common';
+
+export type TransactionExportRow = {
+  type: 'in' | 'out',
+  amount: string,
+  fee: string,
+  date: Date
+}
+
+export type CsvData = {
+  headers: Array<string>,
+  rows: Array<Array<string>>
+}
+
+const TRANSACTION_EXPORT_FILE_TYPE = Object.freeze({
+  csv: 'csv'
+});
+export type TransactionExportFileType = $Values<typeof TRANSACTION_EXPORT_FILE_TYPE>;
+
+const TRANSACTION_EXPORT_DATA_FORMAT = Object.freeze({
+  CoinTracking: 'CoinTracking'
+});
+export type TransactionExportDataFormat = $Values<typeof TRANSACTION_EXPORT_DATA_FORMAT>;
+
+export type ExportFileResponse = {
+  data: Blob,
+  fileType: TransactionExportFileType
+}
+
+/**
+ * This api provides functions to export abstract lists of transactions
+ * as files in different formats and different types.
+ *
+ * Also provides functionality to send abstract byte-blobs as files for user to download.
+ */
+export default class ExportApi {
+
+  /**
+   * Request object MUST contains:
+   * - rows: array of export-data
+   * - fileName: only name without extension
+   *
+   * No result will be returned. File is sent to user as side-effect.
+   */
+  exportTransactions = async (
+    request : ExportTransactionsRequest
+  ): Promise<ExportTransactionsResponse> => {
+    const { rows, format, fileType, fileName } = request;
```

Comment:
@vantuz-subhuman @nicarq 

What is the reason behind passing `fileName` from outside?

User don't have control over file name selection.
Even if `exportTransactions` API accepts `fileName` then it should be optional.

So thinking to make `fileName` optional and add `fileName` creation logic at API side.